### DISCUSSION TITLE
6.2 WMS XML Error

### DISCUSSION THREAD

**Discussion by sidneyh-2** [ua-gist604b-f24/syllabus#37] [General]
In the lecture video, Professor Olsson mentioned the layer preview generating a download XML file but for each error, it would need to be sent to the class or himself to figure out what the issue is. I have copied my error message down below. 

<?xml version="1.0" encoding="UTF-8" standalone="no"?><!DOCTYPE ServiceExceptionReport SYSTEM "https://curly-invention-r4p695j95g9pcp9q7-8080.app.github.dev/geoserver/schemas/wms/1.1.1/WMS_exception_1_1_1.dtd"> <ServiceExceptionReport version="1.1.1" >   <ServiceException>
      java.lang.reflect.UndeclaredThrowableExceptionNo such style handler: format = css
</ServiceException></ServiceExceptionReport>

**Comment by sidneyh-2** [comment]
I also tried doing the QGIS work around but when it connects to Geoserver it does not load any layers from the connection to QGIS.

**Reply by aaryno** [reply]
The message suggests "such style handler: format = css" -- this suggests the css plugin may not be working:

https://github.com/ua-gist604b-f24/6-2-geoserver-osm-styled-sidneyh-2/blob/master/docker-compose.yml#L30

That is perplexing, to be honest. I looked at your repo and you haven't changed that file. We could look at the geoserver logs to see if there are any errors related to that plugin. 

Can you send me the geoserver logs? From your codespace, open the docker extension and stop (if it's not stopped) and start your geoserver container. Then right click and select "View Logs".

**Reply by aaryno** [reply]
FWIW, here is the documentation for the `css` plugin: https://docs.geoserver.org/main/en/user/styling/css/index.html

That extension is required for the styling that this osm-styles repo.

**Reply by aaryno** [reply]
Ok, the logs also indicate an issue with the pregeneralized plugin so for whatever reason neither of the extensions are working. 
In the terminal, type `docker compose down` and then `docker compose up` and show me what it says.

**Reply by aaryno** [reply]
```
geoserver-1  | URL does not exist: https://build.geoserver.org/geoserver/2.24.x/ext-latest/geoserver-2.24-SNAPSHOT-css-plugin-plugin.zip

geoserver-1  | URL does not exist: https://build.geoserver.org/geoserver/2.24.x/ext-latest/geoserver-2.24-SNAPSHOT-feature-pregeneralized-plugin-plugin.zip
```
When I look at https://build.geoserver.org/geoserver/2.24.x/ext-latest/ I see the plugin. It looks like the plugin that the codespace geoserver is trying to download has a double `-plugin` in its name. If you don't have that listed in `docker-compose.yml` then it seems to be a bug in the docker container. 2.24 is a little old. Try changing the geoserver version in docker-compose.yml to 2.27

**Reply by aaryno** [reply]
Heads up to anybody reading this - It looks like 2.27 is NOT available but 2.26 IS. When I run in Sidney's workspace, it does appear that the plugins are downloaded correctly.

**Reply by aaryno** [reply]
Final follow-up -- When we looked at the docker-compose.yml file there was an extra line listing the plugins for css and generalized features that had an extra `-plugin` in the name. Removing that solved the issue.
